### ISSUE TITLE
Support for HTTP Header

### ISSUE THREAD

**Issue by madnight** [cedx/reverse-proxy.js#4]
Could you add support for HTTP Header?
Like: Authorization: Basic

**Comment by cedx** [comment]
What kind of headers do you want to handle through the reverse proxy?

If it is only for the basic or digest authentication headers: I don't think so. It is an authentication mecanism: I think it should be handled by the target application. A such mecanism needs a way to manage a list of credentials, either from the file system nor from a database.

Authentication is a little too big for this project. You should really consider handling basic authentication in your target application.

So I need additional arguments 😃

**Comment by madnight** [comment]
I  thought about the `proxy_set_header` feature that nginx provides to access basic auth protected endpoints:

```nginx
http {
    include       mime.types;
    default_type  application/octet-stream;
    sendfile        on;
    keepalive_timeout  65;
    server {
        listen       1337;
        server_name  localhost;
        location / {
            proxy_pass  http://example.com:80;
            proxy_set_header Authorization "Basic Z29vZHVzZXI6c2VjcmV0cGFzc3dvcmQ=";
        }
    }
}
```

**Comment by cedx** [comment]
I understand better now :) I find this use case quite strange (e.g. make private endpoints public), but it can be a useful addition to this project (a Google search indicates that adding headers to the proxied request is not so exceptional). I will try to implement it but it will take some time as it changes a lot the way I manage the configuration file.

**Comment by cedx** [comment]
Fixed in the released version 2.2.0.

It is a rather basic implementation of this feature : see the updated [documentation](https://github.com/cedx/reverse-proxy.js/blob/master/README.md), or the [http_headers.yaml](https://github.com/cedx/reverse-proxy.js/blob/master/example/yaml/http_headers.yaml) file for a code sample.
